library/Frontend/test/Sexp.hs

```diff
@@ -394,6 +396,40 @@ caseTest =
         \    ((A (B (:record (a) (b)))) (:infix + a b)) \
         \    ((A (B c) (C d))           (:infix + c d))))"
 
+--------------------------------------------------------------------------------
+-- Effects
+--------------------------------------------------------------------------------
+
+handlerTest :: T.TestTree
+handlerTest =
+  T.testGroup
+    "handler sexp parser"
+    [ T.testCase "basic" (basic T.@=? basicExpected)
+    ]
+  where
+    basic =
+      Parser.parse
+        "handler print where op print x = print x , return x = toString x"
+        |> singleEleErr
+    basicExpected =
+      Sexp.parse
+        "(:defhandler print (:ops (:defop print (x) (print x))) (:defop return (x) (toString x)))"
```

Comment:
I see what you are trying to go for, yeah you may want an alist style like
```
(:defhandler print
  :ops ((print (x) (print x)))
  :defop ((return (x) (toString x))))
```

if we can have more than one, if we can only have one then

```lisp

(:defhandler print
  :ops (print (x) (print x))
  :defop (return (x) (toString x)))
```